core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/DataSynchronizer.java

```diff
@@ -1801,6 +1827,11 @@ void insertOne(final MongoNamespace namespace, final BsonDocument document) {
    */
   void insertMany(final MongoNamespace namespace,
                          final List<BsonDocument> documents) {
+    // Remove forbidden fields from the documents before inserting them into the local collection.
+    for (final BsonDocument document : documents) {
+      sanitizeDocument(document);
```

Comment:
mutating a user's data structures would only ever happen when a user is intentionally trying to use the forbidden field, which we will explicitly document not to do. I opted for the mutation so as not to take the performance hit of copying the entire document, which could be up to 16MB.

Though I guess I could have `sanitizeDocument` return a document, and do two passes on the document to avoid mutating: one to check if the forbidden field exists, and another to copy it if the forbidden field is present. In almost every case it will just return the same document reference rather than a copy.